testing/trino-tests/src/test/java/io/trino/tests/TestExcludeColumnsFunction.java

```diff
@@ -0,0 +1,222 @@
+                                    columns => DESCRIPTOR(comment)))
+                """))
+                .matches("SELECT nationkey, name, regionkey FROM tpch.tiny.nation");
+
+        assertThat(query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(regionkey, nationkey)))
+                """))
+                .matches("SELECT name, comment FROM tpch.tiny.nation");
+    }
+
+    @Test
+    public void testInvalidArgument()
+    {
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => CAST(null AS DESCRIPTOR)))
+                """))
+                .hasMessage("Excluded columns descriptor is null");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR()))
+                """))
+                .hasMessage("line 4:21: Invalid descriptor argument COLUMNS. Descriptors should be formatted as 'DESCRIPTOR(name [type], ...)'");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(foo, comment, bar)))
+                """))
+                .hasMessage("Excluded columns: [foo, bar] not present in the table");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(nationkey bigint, comment)))
+                """))
+                .hasMessage("Excluded columns descriptor contains types");
+
+        assertThatThrownBy(() -> query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(tpch.tiny.nation),
+                                    columns => DESCRIPTOR(nationkey, name, regionkey, comment)))
+                """))
+                .hasMessage("All columns are excluded");
+    }
+
+    @Test
+    public void testColumnResolution()
+    {
+        // excluded column names are matched case-insensitive
+        assertThat(query("""
+                SELECT *
+                FROM TABLE(exclude_columns(
+                                    input => TABLE(SELECT 1, 2, 3, 4, 5) t(a, B, "c", "D", e),
+                                    columns => DESCRIPTOR("A", "b", C, d)))
+                """))
+                .matches("SELECT 5");
```

Comment:
As discussed elsewhere, that’s a brittle way to test the output column names of a table function, as it’s looking at the column names coming out of the engine (i.e., whatever is synthesized by the OutputNode at the top of the query plan. In some cases, it may match the column names from the table, but it might not for other cases such as anonymous columns or expressions without aliases)

Reply:
I added a test for that. However please note that we don't follow the ansi SQL semantics for identifiers, and as the result, the column names provided to the function might be already wrong. In your example, the canonical names are "A", "B", but the function gets "a", "B".